vgen/src/main/resources/io/gapi/vgen/go/discovery_fragment.snip

```diff
@@ -0,0 +1,138 @@
+@snippet generateFilename(method)
+  {@method.getName}.frag.go
+@end
+
+# Not used for discovery
+@snippet generateBody(method)
+@end
+
+@snippet generateClass(method, body)
+package main
+
+import (
+  "log"
+
+  "golang.org/x/net/context"
+  "golang.org/x/oauth2/google"
+
+  "google.golang.org/api/{@context.getApi.getName}/{@context.getApi.getVersion}"
+)
+
+func main() {
+  // Use oauth2.NoContext if there isn't a good context to pass in.
+  ctx := context.Background()
+  # Every API except autoscaler has the string as API.CloudPlatformScope, find out why
+  client, err := google.DefaultClient(ctx, "https://www.googleapis.com/auth/cloud-platform")
+  if err != nil {
+    log.Fatal(err)
+  }
+  service, err := {@context.getApi.getName}.New(client)
+  if err != nil {
+    log.Fatal(err)
+  }
+
+  {@initialize(method)}
+
+  @if context.isPageStreaming(method)
+    {@iterableCall(method)}
+  @else
+    {@compactCall(method)}
+  @end
+}
+@end
+
+@private compactCall(method)
+  @if context.isResponseEmpty(method)
+    err = {@methodPath(method)}({@callArg(method)}).Context(ctx).Do()
+    if err != nil {
+      log.Fatal(err)
+    }
+  @else
+    callResult, err := {@methodPath(method)}({@callArg(method)}).Context(ctx).Do()
+    if err != nil {
+      log.Fatal(err)
+    }
+    // doThingsWith(callResult)
+    _ = callResult
+  @end
+@end
+
+@private iterableCall(method)
+  @let responseType = method.getResponseTypeUrl, \
+      responseTypeRef = context.getApiaryConfig.getType(responseType), \
+      resourceField = context.getFirstRepeatedField(responseTypeRef), \
+      resources = context.getResourcesName(resourceField)
+    fn := func(page *{@context.getApi.getName}.{@responseType}) error {
+      // doThingsWith(page)
+      for i, element := range page.{@resources} {
+        // doThingsWith(element)
+        _, _ = i, element
+      }
+      return nil
+    }
+  @end
+
+  err = {@methodPath(method)}({@callArg(method)}).Pages(ctx, fn)
+  if err != nil {
+    log.Fatal(err)
+  }
+@end
+
+@private methodPath(method)
+  service.{@resourcePath(method)}.{@context.lowerCamelToUpperCamel(context.getMethodName(method))}
+@end
+
+@private resourcePath(method)
+  @join r : context.getApiaryConfig.getResources(method.getName) on "."
+    {@context.lowerCamelToUpperCamel(r)}
+  @end
+@end
+
+@private initialize(method)
+  @if context.getMethodParams(method)
+    // TODO: Change placeholders below to values for parameters to the '{@context.getMethodName(method)}' method:
+
+  @end
+  @join param : context.getFlatMethodParams(method)
+    @let signatureType = context.getApiaryConfig.getType(method.getRequestTypeUrl), \
+        paramField = context.getField(signatureType, param), \
+        paramValue = context.typeDefaultValue(signatureType, paramField), \
+        paramDescription = context.getApiaryConfig.getDescription(signatureType.getName, param)
+      @if paramDescription
+        // {@paramDescription}
```

Comment:
@vchudnov-g I think both look a little funny. In the first, it seems odd that we are writing the TODO so many times. In the second, the space separation makes it difficult to see where one variable starts and ends. What about
```
// TODO: Change placeholders...
var (
  // foo signals alpha
  foo = ""

  // bar signals beta
  bar = ""
)
```
?